django/contrib/auth/tokens.py

```diff
@@ -94,7 +94,9 @@ def _make_hash_value(self, user, timestamp):
         # Truncate microseconds so that tokens are consistent even if the
         # database doesn't support microseconds.
         login_timestamp = '' if user.last_login is None else user.last_login.replace(microsecond=0, tzinfo=None)
-        return str(user.pk) + user.password + str(login_timestamp) + str(timestamp)
+        email_field = user.get_email_field_name()
+        email = getattr(user, email_field, '') or ''
+        return str(user.pk) + user.password + str(login_timestamp) + str(timestamp) + email
```

Comment:
It will be more readable with f-string, IMO:
```suggestion
        return f'{user.pk}{user.password}{login_timestamp}{timestamp}{email}'
```